src/components/BookItem.js

```diff
@@ -1,28 +1,33 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/BookSlice';
 
-const BookItem = () => (
-  <div>
-    <li>
-      Atomic Habits
-      <span>James Clear</span>
-      <button type="button">Delete</button>
-    </li>
-    <li>
-      The Spooks secret
-      <span>Joseph Delaney</span>
-      <button type="button">Delete</button>
-    </li>
-    <li>
-      Alice in wonderland
-      <span>Lewis carrol</span>
-      <button type="button">Delete</button>
-    </li>
+const BookItem = ({ book }) => {
+  const { id, title, author } = book;
+  const dispatch = useDispatch();
+
+  return (
     <li>
-      Pride and Prejudice
-      <span>Jane Austen</span>
-      <button type="button">Delete</button>
+      <div className="book">
+        <span className="title">{title}</span>
+        <span className="author">{author}</span>
+      </div>
+      <button
+        type="submit"
+        className="removeBook"
+        onClick={() => dispatch(removeBook(id))}
+      >
+        Remove
+      </button>
     </li>
-  </div>
-);
+  );
+};
+BookItem.propTypes = {
+  book: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  author: PropTypes.string.isRequired,
+};
```

Comment:
- [ ] **[OPTIONAL]** You did a great job working on this milestone! 💯 However, please make sure to define your types correctly. At the moment, this warning is displayed in the console:

![image](https://github.com/stephenkati/book-store/assets/77069011/f36be28c-15f8-4499-9a47-ec5a65aa326a)

and is because according to this propTypes, your `BookItem` expects properties `book, id, title, author`, but in your actual component you're only using the `book` one. Please fix this error! 🤓 🔧